sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxGatewayStoreModel.java

```diff
@@ -299,7 +301,10 @@ private String ensureSlashPrefixed(String path) {
                        }
 
                        if (WebExceptionUtility.isNetworkFailure(dce)) {
-                           BridgeInternal.setSubStatusCode(dce, HttpConstants.SubStatusCodes.GATEWAY_ENDPOINT_UNAVAILABLE);
+                           if (!(dce.getCause() instanceof ReadTimeoutException)) {
```

Comment:
the safer approach is to do a recursive cause check similar to WebExceptionUtility.isNetworkFailure().

Can this be factored out as a utility method to `WebExceptionUtility` similar to `WebExceptionUtility.isNetworkFailure().`?

Reply:
This is done.